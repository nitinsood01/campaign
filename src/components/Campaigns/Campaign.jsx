import React, { useEffect } from "react";
import CampaignsTable from "../../containers/Campaigns/CampaignTable";
import DateRangeFilter from "../../containers/Filters/DateRangeFilter";
import SearchCampaignFilter from "../../containers/Filters/SearchCampaignFilter";

export const Campaign = props => {
  useEffect(() => {
    const { addCampaigns, getUsers } = props;
    window.AddCampaigns = records => {
      addCampaigns(records);
    };
    getUsers();
  }, [props]);
  return (
    <div>
      <div className="position-center flex-dir-row mr-5">
        <DateRangeFilter />
        <SearchCampaignFilter />
      </div>
      <div className="position-center">
        <CampaignsTable />
      </div>
    </div>
  );
};
