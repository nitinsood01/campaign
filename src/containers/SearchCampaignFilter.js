import { connect } from "react-redux";
import { SearchCampaign } from "../components/SearchCampaignFilter.jsx";
import { searchRecords } from "../actions/Campaigns";

export const mapStateToProps = ({ campaignReducer: { searchText } }) => ({
  searchText
});

export const mapDispatchToProps = dispatch => {
  return {
    searchCampaignRecords: text => dispatch(searchRecords(text))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchCampaign);
