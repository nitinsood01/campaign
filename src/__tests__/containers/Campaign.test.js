import renderer from "react-test-renderer";
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { mapDispatchToProps } from "../../containers/Campaign";
import reducers from "../../reducers";
import Campaign from "../../containers/Campaign";

describe("Campaign", () => {
  describe("mapDispatchToProps", () => {
    it("should map dispatch correctly", () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).addCampaigns();
      expect(dispatch.mock.calls[0]).toEqual([
        {
          type: "ADD_CAMPAIGNS",
          data: []
        }
      ]);
    });
  });
  describe("Campaign Table display component", () => {
    it("should not regress", () => {
      const store = createStore(reducers, applyMiddleware(thunk, logger));
      const tree = renderer.create(
        <Provider store={store}>
          <Campaign />
        </Provider>
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
