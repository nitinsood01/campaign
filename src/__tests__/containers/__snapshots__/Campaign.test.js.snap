// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign Campaign Table display component should not regress 1`] = `
<div>
  <div
    className="position-center flex-dir-row mr-5"
  >
    <div>
      <input
        className="m-8"
        max="Invalid date"
        name="start"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Start Date"
        type="text"
        value=""
      />
      <input
        className="m-8"
        min="Invalid date"
        name="end"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="End Date"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        className="m-8"
        onChange={[Function]}
        placeholder="Search by Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="position-center"
  >
    <p>
      Loading...
    </p>
  </div>
</div>
`;
